test/test_thread_safety.rb

```diff
@@ -139,13 +139,13 @@ def test_each_thread_has_its_own_connection_and_can_switch
 
     # Each thread only saw one connection
     threads_to_connections.each do |_thread, connections|
-      assert_equal(1, connections.uniq.length)
+      assert_equal(2, connections.uniq.length)
```

Comment:
I don’t like this. Could we test with some other connections, so we don’t affect this thread safely test?

Reply:
I'm not sure I follow. Previously each thread would have seen only one unique connection proxy (`Pool1DbA.connection` and `Pool1DbB.connection` are `eql?` which is used by `uniq`)  but now it sees two (`Pool1DbA.connection` and `Pool1DbB.connection` are  not `eql?` since the database is considered).

Reply:
Ahh, gotcha. But the comments in line 140 and 146 need to be more specific now. The test was hard to read before, but it’s even harder to read now 😅

Reply:
Very true. Let me know if the updates I made to the comments are on point 🙇